runners/java-fn-execution/src/test/java/org/apache/beam/runners/fnexecution/graph/LengthPrefixUnknownCodersTest.java

```diff
@@ -17,78 +17,122 @@
-import org.apache.beam.sdk.values.KV;
 import org.junit.Test;
 import org.junit.runner.RunWith;
-import org.junit.runners.JUnit4;
+import org.junit.runners.Parameterized;
+import org.junit.runners.Parameterized.Parameter;
+import org.junit.runners.Parameterized.Parameters;
 
 /** Tests for {@link LengthPrefixUnknownCoders}. */
-@RunWith(JUnit4.class)
+@RunWith(Parameterized.class)
 public class LengthPrefixUnknownCodersTest {
-  private static final Coder<WindowedValue<KV<String, Integer>>> windowedValueCoder =
-      WindowedValue.getFullCoder(KvCoder.of(StringUtf8Coder.of(), VarIntCoder.of()),
-          GlobalWindow.Coder.INSTANCE);
 
-  private static final Coder<WindowedValue<KV<String, Integer>>> prefixedWindowedValueCoder =
-      WindowedValue.getFullCoder(KvCoder.of(LengthPrefixCoder.of(StringUtf8Coder.of()),
-          LengthPrefixCoder.of(VarIntCoder.of())), GlobalWindow.Coder.INSTANCE);
+  private static class UnknownCoder extends CustomCoder<String> {
+    private static final Coder<?> INSTANCE = new UnknownCoder();
+    @Override
+    public void encode(String value, OutputStream outStream) throws CoderException, IOException {
+    }
 
-  private static final Coder<WindowedValue<KV<byte[], byte[]>>>
-      prefixedAndReplacedWindowedValueCoder = WindowedValue.getFullCoder(
-      KvCoder.of(
-          LengthPrefixCoder.of(ByteArrayCoder.of()),
-          LengthPrefixCoder.of(ByteArrayCoder.of())),
-      GlobalWindow.Coder.INSTANCE);
+    @Override
+    public String decode(InputStream inStream) throws CoderException, IOException {
+      return "";
+    }
 
-  /** Test wrapping unknown coders with {@code LengthPrefixCoder}. */
-  @Test
-  public void testLengthPrefixUnknownCoders() throws Exception {
-    assertEqualsAfterLengthPrefixingProtoCoder(
-        windowedValueCoder, prefixedWindowedValueCoder, false);
-  }
-
-  /** Test bypassing unknown coders that are already wrapped with {@code LengthPrefixCoder}. */
-  @Test
-  public void testLengthPrefixForLengthPrefixCoder() throws Exception {
-    Coder<WindowedValue<KV<String, Integer>>> windowedValueCoder =
-        WindowedValue.getFullCoder(KvCoder.of(StringUtf8Coder.of(),
-            LengthPrefixCoder.of(VarIntCoder.of())), GlobalWindow.Coder.INSTANCE);
+    @Override
+    public int hashCode() {
+      return 1278890232;
+    }
 
-    Coder<WindowedValue<KV<String, Integer>>> expectedCoder =
-        WindowedValue.getFullCoder(KvCoder.of(LengthPrefixCoder.of(StringUtf8Coder.of()),
-            LengthPrefixCoder.of(VarIntCoder.of())), GlobalWindow.Coder.INSTANCE);
+    @Override
+    public boolean equals(Object obj) {
+      return obj instanceof  UnknownCoder;
+    }
+  }
 
-    assertEqualsAfterLengthPrefixingProtoCoder(windowedValueCoder, expectedCoder, false);
+  @Parameters
+  public static Collection<Object[]> data() {
+    return ImmutableList.of(
+        /** Test wrapping unknown coders with {@code LengthPrefixCoder}. */
+        new Object[] {
```

Comment:
I prefer the use of an `AutoValue` inner class here instead of an `Object[]`

Reply:
I'm going to stick with the Object[] for now since it will make the migration to JUnit 5 parameterized tests simpler.